plugins/FeatureFlags/Commands/ManageFeatureFlags.php

```diff
@@ -0,0 +1,97 @@
+use Piwik\Plugin\ConsoleCommand;
+use Piwik\Plugins\FeatureFlags\FeatureFlagInterface;
+use Piwik\Plugins\FeatureFlags\FeatureFlagStorageInterface;
+
+class ManageFeatureFlags extends ConsoleCommand
+{
+    private const ACTION_OPTIONS = [
+        'enable',
+        'disable',
+        'delete'
+    ];
+
+    /**
+     * This method allows you to configure your command. Here you can define the name and description of your command
+     * as well as all options and arguments you expect when executing it.
+     */
+    protected function configure()
+    {
+        $this->setName('featureflags:manage');
+        $this->setDescription('Manage feature flags (enable/disable/delete)');
+        $this->addRequiredValueOption('name', null, 'Feature flag name');
+        $this->addRequiredValueOption('action', null, 'enable/disable/delete');
+    }
+
+    /**
+     * The actual task is defined in this method. Here you can access any option or argument that was defined on the
+     * command line via $this->getInput() and write anything to the console via $this->getOutput().
+     * In case anything went wrong during the execution you should throw an exception to make sure the user will get a
+     * useful error message and to make sure the command does not exit with the status code 0.
+     *
+     * Ideally, the actual command is quite short as it acts like a controller. It should only receive the input values,
+     * execute the task by calling a method of another class and output any useful information.
+     *
+     * Execute the command like: ./console examplecommand:helloworld --name="The Matomo Team"
+     */
+    protected function doExecute(): int
+    {
+        $input = $this->getInput();
+
+        $name = $input->getOption('name');
+        $action = $input->getOption('action');
+
+        if (!in_array($action, self::ACTION_OPTIONS)) {
+            throw new \Exception("Action must be 'enable', 'disable' or 'delete'");
+        }
+
+        $featureFlag = $this->loadFeatureFlag($name);
+
+        if ($featureFlag === null) {
+            throw new \Exception("Feature flag could not be found");
+        }
+
+        /** @var FeatureFlagStorageInterface $storage */
+        foreach (StaticContainer::get('featureflag.storages') as $storage) {
+            switch ($action) {
+                case self::ACTION_OPTIONS[0]: // Enable
+                    $storage->enableFeatureFlag($featureFlag);
+                    break;
+                case self::ACTION_OPTIONS[1]: // Disable
+                    $storage->disableFeatureFlag($featureFlag);
+                    break;
+                case self::ACTION_OPTIONS[2]: // Delete
+                    $storage->deleteFeatureFlag($featureFlag);
+                    break;
+            }
+        }
+
+        return self::SUCCESS;
+    }
+
+    private function loadFeatureFlag(string $name): ?FeatureFlagInterface
+    {
+        $classes = StaticContainer::get('featureflag.feature_flags');
```

Comment:
Have you thought about using `findMultipleComponents` from the [Plugin Manager](https://github.com/matomo-org/matomo/blob/15b88846fe40c2be07b9b42c86de2f973d8ab676/core/Plugin/Manager.php#L617)?

It would enforce some part of the directory structure and remove the DI configuration requirement to find them:

```suggestion
        $classes = \Piwik\Plugin\Manager::getInstance()->findMultipleComponents('FeatureFlags', FeatureFlagInterface::class);
```

This would only make sense if we also remove the DI check inside `FeatureFlagManager::createFeatureFlagObjFromString()`. I would argue it should not be necessary there at all as the check for the interface implementation should be enough :thinking:

Reply:
Sounds like a better idea, I've used that now and your last comment is right as well.

The benefit being you don't need a predefined list of feature flags, the only challenge with that is regarding testing, it looks in a specific directory only, so I've made the directory it looks in configurable so the test feature flags and regular don't get bunched together.